feature_engine/datetime/date.py

```diff
@@ -0,0 +1,169 @@
+        Note: if you pass a subset of features via the variables argument
+        of the transformer and drop_datetime=True, there might be more
+        datetime features in the dataframe that will not be dropped
+        upon calling the transform method
+
+    kwargs:
+        see pd.to_datetime keyword arguments other than "errors"
+
+    Attributes
+    ----------
+    supported:
+        list of supported date features
+
+    Available upon calling the fit method:
+
+    variables_:
+        List of variables from which to extract date features
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Find the variables from which to extract date features
+    transform:
+        Add the new date features specified in features_to_extract argument
+    fit_transform:
+        Fit to the data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        features_to_extract: Union[str, List[str]] = "year",
+        drop_datetime: bool = True,
+        **kwargs
+    ) -> None:
+
+        # get the list of supported features from a const variable somewhere?
+        self.supported = [
+            "month",
+            "quarter",
+            "semester",
+            "year",
+            "week_of_the_year",
+            "day_of_the_week",
+            "day_of_the_month",
+            "is_weekend",
+            "week_of_the_month",
+        ]
+
+        if not isinstance(features_to_extract, (str, list)):
+            raise TypeError("features_to_extract must be either a str or a list of str")
+
+        if features_to_extract == "all":
+            features_to_extract = self.supported
+
+        elif isinstance(features_to_extract, str):
+            features_to_extract = [features_to_extract]
+
+        if any(feature not in self.supported for feature in features_to_extract):
+            raise ValueError(
+                "At least one of the requested feature is not supported. "
+                "Supported features are {}.".format(", ".join(self.supported))
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.drop_datetime = drop_datetime
+        self.kwargs = kwargs
+        self.features_to_extract = features_to_extract
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        X = super().fit(X, y)
+        self.n_features_in_ = X.shape[1]
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+
+        X = _is_dataframe(X)
+        X = super().transform(X)
+
+        # maybe iterate the following with func handles or smth
+        # maybe rearrange final columns so that features extracted from
+        # the same variable are grouped together
+
+        if "month" in self.features_to_extract:
+            for var in self.variables_:
```

Comment:
I see you put the loop in, it generates boiler plate, if you take this line out of the loop, then you don't need to re-write it so many times. And it will also generate that the features from the same variable are one next to the other in the final df. Easier to understand for the user i think

Reply:
Yea no this was terrible, I don't know what I was thinking